daemon/daemon_unix.go

```diff
@@ -1702,51 +1699,32 @@ func setupOOMScoreAdj(score int) error {
 	return err
 }
 
-func (daemon *Daemon) initCgroupsPath(path string) error {
+func (daemon *Daemon) initCPURtController(mnt, path string) error {
 	if path == "/" || path == "." {
 		return nil
 	}
 
-	if daemon.configStore.CPURealtimePeriod == 0 && daemon.configStore.CPURealtimeRuntime == 0 {
-		return nil
-	}
-
-	if cgroups.IsCgroup2UnifiedMode() {
-		return fmt.Errorf("daemon-scoped cpu-rt-period and cpu-rt-runtime are not implemented for cgroup v2")
-	}
-
 	// Recursively create cgroup to ensure that the system and all parent cgroups have values set
 	// for the period and runtime as this limits what the children can be set to.
```

Comment:
Unrelated question; if we recursively set this cgroup, does each recursion limit the rc period/runtime further as a part of its parent? (IoW, are they cummulative?) So if we would set the daemon's `--cgroup-parent=/one/two/three`, then the result would be different than `--cgroup-parent=/one` ? Because in that case, we should only set the cgroup _once_ (for the docker "root" cgroup)? (or is it not possible to "skip" parent cgroups?)

Reply:
That's a very good question that I am also curious about (and this is that time that I wish I could have asked it during the review of #23430).

Some docs are available at https://www.kernel.org/doc/html/latest/scheduler/sched-rt-group.html

From what I understand, this is not a limit but rather a hard allocation (meaning the cgroup is given that amount of CPU time exclusively). So, you can think of it as a disk space -- if your want to have 10 GB free in /some/fancy/dir, you have to have it in its parent (/some/fancy) as well.

The idea of making dockerd set this for all parent cgroups seems questionable to me, but now we have what we have, and removing this will surely break backward compatibility.